Box.V2/Managers/BoxResourceManager.cs

```diff
@@ -257,6 +257,36 @@ protected async Task<BoxCollectionMarkerBased<T>> AutoPaginateMarkerMetadataQuer
             return allItemsCollection;
         }
 
+        /// <summary>
+        /// Used to fetch all results using pagination for metadata queries. V2 is for when a BoxItem is returned when fields are passed in.
+        /// </summary>
+        /// <typeparam name="T">The type of BoxCollectionMarkerBased item to expect.</typeparam>
+        /// <param name="request">The pre-configured BoxRequest object.</param>
+        /// <returns></returns>
+        protected async Task<BoxCollectionMarkerBased<T>> AutoPaginateMarkerMetadataQueryV2<T>(BoxRequest request) where T : BoxItem, new()
```

Comment:
```suggestion
        protected async Task<BoxCollectionMarkerBased<T>> AutoPaginateMarkerMetadataQueryForBoxItems<T>(BoxRequest request) where T : BoxItem, new()
```

Reply:
Oh man, "V2" is not very descriptive.  Is this the pattern for this SDK in this case?  What if V2 was replaced with "ForBoxItems" or something similar?  Otherwise, I understand keeping with the SDK naming convention.

Reply:
Ya it seems like the naming convention in the SDK. Also, we will be taking it out once metadata query is fixed to only provide one of these formats.